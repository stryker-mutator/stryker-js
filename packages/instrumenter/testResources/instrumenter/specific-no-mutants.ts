const a = 1 + 1;
const b = 1 - 1;
