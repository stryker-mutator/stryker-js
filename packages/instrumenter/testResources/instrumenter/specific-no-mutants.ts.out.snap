// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`instrumenter integration with mutation ranges should not make any mutations in a file not found in the specific mutants 1`] = `
"function stryNS_9fa48() {
  var g = new Function(\\"return this\\")();
  var ns = g.__stryker__ || (g.__stryker__ = {});

  if (ns.activeMutant === undefined && g.process && g.process.env && g.process.env.__STRYKER_ACTIVE_MUTANT__) {
    ns.activeMutant = g.process.env.__STRYKER_ACTIVE_MUTANT__;
  }

  function retrieveNS() {
    return ns;
  }

  stryNS_9fa48 = retrieveNS;
  return retrieveNS();
}

stryNS_9fa48();

function stryCov_9fa48() {
  var ns = stryNS_9fa48();
  var cov = ns.mutantCoverage || (ns.mutantCoverage = {
    static: {},
    perTest: {}
  });

  function cover() {
    var c = cov.static;

    if (ns.currentTestId) {
      c = cov.perTest[ns.currentTestId] = cov.perTest[ns.currentTestId] || {};
    }

    var a = arguments;

    for (var i = 0; i < a.length; i++) {
      c[a[i]] = (c[a[i]] || 0) + 1;
    }
  }

  stryCov_9fa48 = cover;
  cover.apply(null, arguments);
}

function stryMutAct_9fa48(id) {
  var ns = stryNS_9fa48();

  function isActive(id) {
    if (ns.activeMutant === id) {
      if (ns.hitCount !== void 0 && ++ns.hitCount > ns.hitLimit) {
        throw new Error('Stryker: Hit count limit reached (' + ns.hitCount + ')');
      }

      return true;
    }

    return false;
  }

  stryMutAct_9fa48 = isActive;
  return isActive(id);
}

const a = stryMutAct_9fa48(\\"0\\") ? 1 - 1 : (stryCov_9fa48(\\"0\\"), 1 + 1);
const b = stryMutAct_9fa48(\\"1\\") ? 1 + 1 : (stryCov_9fa48(\\"1\\"), 1 - 1);"
`;

exports[`instrumenter integration with mutation ranges should not make any mutations in a file with mutate = [] 1`] = `
"const a = 1 + 1;
const b = 1 - 1;"
`;
